openfeature/api.go

```diff
@@ -211,6 +193,31 @@ func (api *evaluationAPI) initNewAndShutdownOld(newProvider FeatureProvider, old
 	}(v)
 }
 
+// initializer is a helper to execute provider initialization and generate appropriate event for the initialization
+func initializer(provider FeatureProvider, apiCtx EvaluationContext) Event {
+	var event = Event{
+		ProviderName: provider.Metadata().Name,
+		EventType:    ProviderReady,
+		ProviderEventDetails: ProviderEventDetails{
+			Message: "Provider initialization successful",
+		},
+	}
+
+	handler, ok := provider.(StateHandler)
+	if !ok {
+		// Note - a provider without state handling capability can be assumed to be ready immediately.
+		return event
+	}
+
+	err := handler.Init(apiCtx)
+	if err != nil {
+		event.EventType = ProviderError
+		event.Message = fmt.Sprintf("Provider initialization error, %v", err)
```

Comment:
I think we should also clean up the `ProviderEventDetails` property otherwise it will be quite weird to fail successfully 🤣

Reply:
![image](https://github.com/open-feature/go-sdk/assets/25272906/7642b0b8-e021-409b-baad-d59dffd2fc8c)

Reply:
@thisthat  good concern & @toddbaert  nice one 😅 

Luckily in OF go-sdk, this is handled correctly as we use go struct embedding [1] where `message` updated in error is from `ProviderEventDetails`

![image](https://github.com/open-feature/go-sdk/assets/8186721/0a17ad30-a379-4d7a-852d-954cd3315e10)

I know this is little confusing but I wanted to avoid multiple event initializations and went ahead with this. Let me know if you think we need to change this :) 

[1] - https://github.com/open-feature/go-sdk/blob/main/openfeature/provider.go#L105-L109

Reply:
oohh gotcha! All good for me then 👍

Reply:
I always forget about struct embedding!